aea/manager.py

```diff
@@ -664,9 +664,12 @@ def _make_config(
             )
             component_id = ComponentId(obj.pop("type"), PublicId(author, name, version))
             components_configs[component_id] = obj
-        agent_update_dictionary["component_configurations"] = components_configs
-        # do the override (and valiation)
-        agent_config.update(agent_update_dictionary)
+
+        # update components
+        agent_config.update({"component_configurations": components_configs})
+        json_data = agent_config.ordered_json
+        json_data.update(agent_update_dictionary)
+        agent_config = AgentConfig.from_json(json_data)
```

Comment:
why are these three lines needed, why is update in line  not enough? It looks like the update method of `AgentConfig` should be extended if anything is missing rather than this weird approach here.

Reply:
@DavidMinarsch 
yes, update method is incomplete, and the proper fix requires better design and will take some time.